gpflow/utilities/utilities.py

```diff
@@ -247,18 +251,133 @@ def clear_bijector(bijector, _, state):
     return input_module
 
 
+def _get_by_name_index(parent: object, attr_str: str, index_str: str) -> object:
+    attr = getattr(parent, attr_str)
+    if index_str is not None:
```

Comment:
should the type annotation be `Optional[str]` then?

Reply:
Nope, but it should be another type.